compiler/qsc_fir/src/global.rs

```diff
@@ -5,12 +5,15 @@ use crate::{
     fir::{Item, ItemId, ItemKind, Package, PackageId, Visibility},
     ty::Scheme,
 };
-use qsc_data_structures::index_map;
+use qsc_data_structures::{
+    index_map,
+    namespaces::{NamespaceId, NamespaceTreeRoot},
+};
 use rustc_hash::FxHashMap;
 use std::rc::Rc;
 
 pub struct Global {
-    pub namespace: Rc<str>,
+    pub namespace: Vec<Rc<str>>,
```

Comment:
A `NamespaceTreeRoot` _sounds like_ the kind of type this should be? I'm surprised to see a vector of strings here after the addition of other namespace types.